packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -227,6 +234,27 @@ export async function activate(context: vscode.ExtensionContext) {
     sfdxProjectOpened = files && files.length > 0;
   }
 
+  // Set environment variable to add logging for VSCode API calls
+  process.env.SFDX_SET_CLIENT_IDS = CLIENT_ID;
+  const config = vscode.workspace.getConfiguration();
+  config.update(
+    TERMINAL_INTEGRATED_ENV_MAC,
+    { SFDX_SET_CLIENT_IDS: CLIENT_ID },
+    ConfigurationTarget.Global
```

Comment:
This will add the environment variable to integrated terminals before they are started.

Reply:
What does the user workspace.json look like when this is added? Can we also add some `// This was added for logging...` in the json file? I don't want to just surprise them.

Also, this is idempotent, right? It will only add it if it is not there. And subsequent runs will not re-populate?

Reply:
It does not look like there is a way to add a comment through the update method since it can only update a known section key.

Thought about it further and this will not only update that field. Unfortunately, we cannot specify updating only SFDX_SET_CLIENT_IDS because this must be registered. I did not take into account that the user might have their own env vars set in this as well. I will need to get their current configuration and append our env var to it. If the user changed SFDX_SET_CLIENT_IDS to something else, do we respect that or do we override it with ours?

Reply:
It's OK to override `SFDX_SET_CLIENT_IDS` with just ours. That is not a generally used variable and changing it doesn't have any effects on functionality.